amplify-ui-geo-explore/src/components/places/PlacesLayer.js

```diff
@@ -0,0 +1,100 @@
+
+// Override default marker icon
+const icon = document.createElement("div");
+icon.innerHTML = renderToString(<PinIcon size={ICON_SIZE} color={"rgb(0, 0, 0)"} />);
+const markerIcon = { element: icon, offset: [0, -ICON_SIZE/2] };
+
+// Override search box's default style
+const render = (item) => {
+  // Render as a suggestion
+  if (typeof item === "string") {
+    let suggestionName = item.split(",");
+    let indexOfMatch = suggestionName[0].toLowerCase().indexOf(this.query.toLowerCase());
+    let lengthOfMatch = this.query.length;
+    let beforeMatch = suggestionName[0].substring(0, indexOfMatch);
+    let match = suggestionName[0].substring(indexOfMatch, indexOfMatch + lengthOfMatch);
+    let afterMatch = suggestionName[0].substring(indexOfMatch + lengthOfMatch);
+    return renderToString(
+      <div className="mapboxgl-ctrl-geocoder--suggestion maplibregl-ctrl-geocoder--suggestion">
+        <div className="mapboxgl-ctrl-geocoder--suggestion-info maplibregl-ctrl-geocoder--suggestion-info">
+          <div className="mapboxgl-ctrl-geocoder--suggestion-title maplibregl-ctrl-geocoder--suggestion-title">
+            {beforeMatch}
+            <span className="mapboxgl-ctrl-geocoder--suggestion-match maplibregl-ctrl-geocoder--suggestion-match">
+              {match}
+            </span>
+            {afterMatch}
+          </div>
+          <div className="mapboxgl-ctrl-geocoder--suggestion-address maplibregl-ctrl-geocoder--suggestion-address">
+            {suggestionName.splice(1, suggestionName.length).join(",")}
+          </div>
+        </div>
+      </div>
+    );
+  } else {
+    // render as a search result
+    let placeName = item.place_name.split(",");
+    return renderToString(
+      <div className="mapboxgl-ctrl-geocoder--result maplibregl-ctrl-geocoder--result">
+        <div>
+          <div className="mapboxgl-ctrl-geocoder--result-title maplibregl-ctrl-geocoder--result-title">
+            {placeName[0]}
+          </div>
+          <div className="mapboxgl-ctrl-geocoder--result-address maplibregl-ctrl-geocoder--result-address">
+            {placeName.splice(1, placeName.length).join(",")}
+          </div>
+        </div>
+      </div>
+    );
+  }
+};
+
+// Override default popup
+const popupRender = (item) => {
+  let placeName = item.place_name.split(",");
+  return renderToString(
+    <div className={styles.popup}>
+      <div className={styles.popup__content}>
+        <div className={styles.popup__title}>{placeName[0]}</div>
+        {placeName.splice(1, placeName.length).join(",")}
+        <div className={styles.popup__coordinates}>
+          {`${item.center[1].toFixed(6)}, ${item.center[0].toFixed(6)}`}
+        </div>
+      </div>
+    </div>
+  );
+};
+
+// Layer in the app that contains Places functionalities
+const PlacesLayer = () => {
+  return (
+    <Geocoder
+      position="top-left"
+      placeholder="Search Places"
+      showResultsWhileTyping={false}
```

Comment:
Why is this set to `false`? While using it I was a bit confused as I didn't know I had to press "enter" and waited a few seconds, and nothing happened.